plugins/UsersManager/UsersManager.php

```diff
@@ -173,13 +174,23 @@ public static function checkPassword($password)
         Piwik::postEvent('UsersManager.checkPassword', array($password));
 
         if (!self::isValidPasswordString($password)) {
+            if ($showError) {
+                return Piwik::translate('UsersManager_ExceptionInvalidPassword', array(self::PASSWORD_MIN_LENGTH));
+            }
             throw new Exception(Piwik::translate('UsersManager_ExceptionInvalidPassword',
-              array(self::PASSWORD_MIN_LENGTH)));
+                array(self::PASSWORD_MIN_LENGTH)));
+
         }
         if (mb_strlen($password) > self::PASSWORD_MAX_LENGTH) {
+            if ($showError) {
+                return Piwik::translate('UsersManager_ExceptionInvalidPasswordTooLong',
+                    array(self::PASSWORD_MAX_LENGTH));
+            }
             throw new Exception(Piwik::translate('UsersManager_ExceptionInvalidPasswordTooLong',
-              array(self::PASSWORD_MAX_LENGTH)));
+                array(self::PASSWORD_MAX_LENGTH)));
+
         }
+        return null;
```

Comment:
To avoid duplicating the translation method calls, throw and return statements this can be simplified to:

```php
        $error = null;

        if (!self::isValidPasswordString($password)) {
            $error = Piwik::translate('UsersManager_ExceptionInvalidPassword', array(self::PASSWORD_MIN_LENGTH));
        }
        else if (mb_strlen($password) > self::PASSWORD_MAX_LENGTH) {
            $error = Piwik::translate('UsersManager_ExceptionInvalidPasswordTooLong', array(self::PASSWORD_MAX_LENGTH));
        }

        if ($error && !$showError) {
            throw new Exception($error);
        }

        return $error;
```